foursight_core/es_connection.py

```diff
@@ -162,7 +162,11 @@ def search(self, search, key='_source') -> Tuple[list, int]:
             raise ElasticsearchException(message=err_msg)
         # In next line, PyCharm's linter wrongly worries that 'res' might not be reliably set above. -kmp 6-Jun-2022
         else:
-            total = res['hits']['total']['value']  # noQA (see above)
+            total = res['hits']['total']  # noQA (see above)
+            # As of ES7 we have result.hits.total.value rather than result.hits.total.
+            # total = res['hits']['total']['value']  # noQA (see above)
+            if not isinstance(total, int):
+                total = total['value']
```

Comment:
The way I recommend addressing this is by a utility function that gets the total value.  I thought Will actually had such a function, but even if you duplicate something, then it reduces the need to do low-level, temporary, custom logic in the middle of something that's doing something else. So I'd expect here to see:
```
total = get_hits_total(res)
```
and then have the function sort it out.

Reply:
Good point.